File: WWF/Hands on labs 3.0/Lab07/Completed/Exercise 1/WorkflowConsoleApplication1/WorkflowConsoleApplication1/Workflow1.cs
//--------------------------------------------------------------------------------
// This file is a "Sample" as from Windows Workflow Foundation
// Hands on Labs RC
//
// Copyright (c) Microsoft Corporation. All rights reserved.
//
// This source code is intended only as a supplement to Microsoft
// Development Tools and/or on-line documentation.  See these other
// materials for detailed information regarding Microsoft code samples.
// 
// THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//--------------------------------------------------------------------------------

using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;

namespace WorkflowConsoleApplication1
{
	public sealed partial class Workflow1: SequentialWorkflowActivity
	{
		public Workflow1()
		{
			InitializeComponent();
		}


        public enum NumberIs { Odd = 0, Even = 1 }
        private NumberIs currentNumber = NumberIs.Even;
        private bool userRequestsExit;

        //number entered is odd
        public void codeActivity1_ExecuteCode(object sender, EventArgs e)
        {
            Console.WriteLine("the number is odd");
            GetNextNumber();
        }

        //number entered is even
        public void codeActivity2_ExecuteCode(object sender, EventArgs e)
        {
            Console.WriteLine("the number is even");
            GetNextNumber();
        }


        //evaluate number entered by user and set currentNumber variable
        private void GetNextNumber()
        {
            int numIn;

            Console.WriteLine("enter a number.");
            string charIn = Console.ReadLine();

            if (Int32.TryParse(charIn, out numIn))
            {
                if ((numIn % 2) == 0)
                {
                    currentNumber = NumberIs.Even;
                }
                else
                {
                    currentNumber = NumberIs.Odd;
                }
            }
            else if (charIn.Contains("x"))
            {
                userRequestsExit = true;
            }
            else
            {
                GetNextNumber();
            }
        }

        //used to control execution of codeActivity1_ExecuteCode1
        public void oddCondition(object sender, ConditionalEventArgs args)
        {
            args.Result = ((currentNumber == NumberIs.Odd));
        }

        //used to control execution of codeActivity2_ExecuteCode1
        public void evenCondition(object sender, ConditionalEventArgs args)
        {
            args.Result = ((currentNumber == NumberIs.Even));
        }

        //used to control execution of the ConditionedActivityGroup
        public void timeToExit(object sender, ConditionalEventArgs args)
        {
            args.Result = userRequestsExit;
        }

	}
}

File: WWF/Hands on labs 3.0/VB/Lab07/Completed/Exercise1/WorkflowConsoleApplication1/WorkflowConsoleApplication1/Workflow1.vb
'--------------------------------------------------------------------------------
' This file is a "Sample" as from Windows Workflow Foundation
' Hands on Labs RC
'
' Copyright (c) Microsoft Corporation. All rights reserved.
'
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'--------------------------------------------------------------------------------

Public class Workflow1
    Inherits SequentialWorkflowActivity

    Public Enum NumberIs
        Odd = 0
        Even = 1
    End Enum
    Private currentNumber As NumberIs = NumberIs.Odd
    Private userRequestsExit As Boolean

    'number entered is odd
    Public Sub codeActivity1_ExecuteCode(ByVal sender As Object, ByVal e As EventArgs)

        Console.WriteLine("the number is odd")
        GetNextNumber()

    End Sub

    'number entered is even
    Public Sub codeActivity2_ExecuteCode(ByVal sender As Object, ByVal e As EventArgs)

        Console.WriteLine("the number is even")
        GetNextNumber()
    End Sub


    'evaluate number entered by user and set currentNumber variable
    Private Sub GetNextNumber()

        Dim numIn As Integer

        Console.WriteLine("enter a number.")
        Dim charIn As String = Console.ReadLine()

        If Int32.TryParse(charIn, numIn) Then

            If ((numIn Mod 2) = 0) Then

                currentNumber = NumberIs.Even

            Else

                currentNumber = NumberIs.Odd
            End If

        ElseIf (charIn.Contains("x")) Then

            userRequestsExit = True

        Else

            GetNextNumber()
        End If
    End Sub


    'used to control execution of codeActivity1_ExecuteCode1
    Public Sub oddCondition(ByVal sender As Object, ByVal args As ConditionalEventArgs)

        args.Result = ((currentNumber = NumberIs.Odd))

    End Sub

    'used to control execution of codeActivity2_ExecuteCode1
    Public Sub evenCondition(ByVal sender As Object, ByVal args As ConditionalEventArgs)

        args.Result = ((currentNumber = NumberIs.Even))

    End Sub

    'used to control execution of the ConditionedActivityGroup
    Public Sub timeToExit(ByVal sender As Object, ByVal args As ConditionalEventArgs)

        args.Result = userRequestsExit

    End Sub



End Class
